dev/tasks/java-jars/github.yml

```
@@ -83,7 +83,7 @@ jobs:
           - { runs_on: ["macos-13"], arch: "x86_64"}
           - { runs_on: ["macos-14"], arch: "aarch_64" }
     env:
-      MACOSX_DEPLOYMENT_TARGET: "10.15"
+      MACOSX_DEPLOYMENT_TARGET: "13.10"
```

Could you add this comment for reference?
```suggestion
      # Current oldest supported version according to https://endoflife.date/macos
      MACOSX_DEPLOYMENT_TARGET: "13.10"
```

https://github.com/apache/arrow/blob/a88f0cd0368f81f1b9243d443cd571d09242c5b1/dev/tasks/verify-rc/github.macos.yml#L24-L25

https://github.com/apache/arrow/blob/a88f0cd0368f81f1b9243d443cd571d09242c5b1/.github/workflows/python.yml#L166-L167

I think these values should be at least `12.7.5`.

---

@llama90 I am actually not sure if your PR caused this, I am merely testing a few ideas. Honestly still looking for a solution :slightly_smiling_face: 
Really appreciate pinging here :+1: